File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;

        for (int i = 0; i < nums.size(); i++) {
            int new_target = target - nums[i];
            if (map.contains(new_target)) {
                return {map[new_target], i};
            }

            map[nums[i]] = i;
        }

        return { };
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();

        for(int i = 0; i < nums.length; i++) {
            int currTarget = target - nums[i];

            if(map.containsKey(currTarget)) {
                return new int[] {map.get(currTarget), i};
            }

            map.put(nums[i], i);
        }

        return new int[] {};
        
    }
}